### DISCUSSION TITLE
resetField generating unnecessary renderings

### DISCUSSION THREAD

**Discussion by javadev-jef** [react-hook-form/react-hook-form#10306] [Q&A] [answered]
Hello, I would like some help with `resetField`. I performed a test with `React Dev Tools` and identified that when the field specified in `resetField` is reset, all form components are also being rendered again, not just the reset component. Is this really expected when executing `resetField`?

I performed this test directly on the https://codesandbox.io/s/priceless-firefly-d0kuv example provided in the documentation.

**Comment by leapful** [comment, marked as answer]
@javadev-jef 

Yes, that's the expected behavior. Please check the documentation here https://react-hook-form.com/api/useform/resetfield/

>After invoke this function.
isValid form state will be reevaluated.
isDirty form state will be reevaluated.

Since form state is being re-evaluated, `useForm` hook also triggers re-rendering following to updated form state.

**Reply by javadev-jef** [reply]
Understood. Thanks @leapful.
